### ISSUE TITLE
Move jackson to separate module

### ISSUE THREAD

**Issue by slamdev** [crnk-project/crnk-framework#80]
It would be nice to have an option to use something else for json processing. 
There are [plenty of tools](https://github.com/fabienrenaud/java-json-benchmark#purpose) which can be used for it.
Other implementations can be added easily when jackson is moved to separate module.

**Comment by remmeier** [comment]
Yes, in general that would be quite well possible. The dependency to Jackson has actually already been substantially reduced the past few months. There are two main places left:

- serialization of Resource, Document (low effort)
- AnnotationResourceInformationBuilder (medium effort)

Next to that there a few references to ObjectMapper like in CrnkBoot that need to be generalized.

**Comment by remmeier** [comment]
https://github.com/crnk-project/crnk-framework/issues/96 introduced a JacksonResourceFieldInformationProvider and a JacksonModule that provide a big step into the area here as well. 

In a next step the remaining Jackson logic would have to be moved to that module and a "SerializerModule" interface/base class introduced. From there it would then be straighforward to introduce other JSON libraries. Contributions arewelcomed .
